### ISSUE TITLE
Testing of other evaluation metrics and target detectionm 其他评估指标的测试及目标检测

### ISSUE THREAD

**Issue by szw811** [XinyiYing/MoCoPnet#5]
感谢您的工作！想问一下，后续是否会开源包含SNRG、BSF、ROC等评估指标的测试代码以及目标检测代码，想作为参考。
再次感谢您的工作，给我带来了很多启发。

Thanks for your work! I would like to ask whether the testing code containing evaluation metrics such as SNRG, BSF, and ROC and target detection code will be open sourced in the future. I would like to use it as a reference.
Thanks again for your work which has inspired me a lot.

**Comment by XinyiYing** [comment]
等有空应该会整理上传，要是两周后没传发邮件提醒我一下

**Comment by szw811** [comment]
> 等有空应该会整理上传，要是两周后没传发邮件提醒我一下

好的，感谢您的工作！祝工作和生活顺利！

**Comment by XinyiYing** [comment]
评估指标的测试代码（LSNRG,BSF,SCR,CR）以及目标检测代码（tophat,ILCM,IPI的公开源码）已经上传 [Link](https://github.com/XinyiYing/MoCoPnet/tree/main/code/supp)

**Comment by szw811** [comment]
非常感谢！！！

**Comment by MuGeeee** [comment]
非常感兴趣您的工作，请问ROC的评估代码可以分享一下吗？

**Comment by MC1016** [comment]
> 评估指标的测试代码（LSNRG,BSF,SCR,CR）以及目标检测代码（tophat,ILCM,IPI的公开源码）已经上传 [Link](https://github.com/XinyiYing/MoCoPnet/tree/main/code/supp)

您好,在您提供的评估代码中,输入需要目标中心坐标,这个在提供的test数据中没有给出

**Comment by XinyiYing** [comment]
[SAITD](https://www.scidb.cn/en/detail?dataSetId=808025946870251520&dataSetType=journal)，[Hui](https://www.scidb.cn/en/detail?dataSetId=720626420933459968&dataSetType=journal)， [Anti-UAV](https://anti-uav.github.io/dataset/) 数据集否提供了目标的真值位置。具体参考各个数据集的详细标注。

**Comment by MC1016** [comment]
> [SAITD](https://www.scidb.cn/en/detail?dataSetId=808025946870251520&dataSetType=journal)，[Hui](https://www.scidb.cn/en/detail?dataSetId=720626420933459968&dataSetType=journal)， [Anti-UAV](https://anti-uav.github.io/dataset/) 数据集否提供了目标的真值位置。具体参考各个数据集的详细标注。

谢谢,在复现过程中还有一些问题麻烦您,SNR和CR也是基于区域邻域计算的,这两个指标在代码里我没有找到,是需要对整个测试集进行计算吗? 此外对于检测的实验部分,评价指标的计算原文是只选择了两张图片吗?

**Comment by XinyiYing** [comment]
论文中关于小目标增强的评测都是基于区域邻域计算的，包括SNR gain (SNRG)和contrast gain (CG)，具体公式见论文公式6和公式9。
关于测试集的具体说明在论文的IV-A-1) Datasets，
In this paper, we employ the 1st−50th sequences with target annotations of SAITD as the test datasets and the remaining 300 sequences as the training datasets. In addition, we employ Hui and Anti-UAV as the test dataset to test the robustness of our MoCoPnet to real scenes. In Anti-UAV dataset, only the sequences with infrared small target (21 sequences in total) are selected as the test set. Note that, we only use the first 100 images of each sequence for test to balance computational/time cost and generalization performance.

**Comment by MC1016** [comment]
> 论文中关于小目标增强的评测都是基于区域邻域计算的，包括SNR gain (SNRG)和contrast gain (CG)，具体公式见论文公式6和公式9。 关于测试集的具体说明在论文的IV-A-1) Datasets， In this paper, we employ the 1st−50th sequences with target annotations of SAITD as the test datasets and the remaining 300 sequences as the training datasets. In addition, we employ Hui and Anti-UAV as the test dataset to test the robustness of our MoCoPnet to real scenes. In Anti-UAV dataset, only the sequences with infrared small target (21 sequences in total) are selected as the test set. Note that, we only use the first 100 images of each sequence for test to balance computational/time cost and generalization performance.

感谢您的回复,这里我指的是在文章中表1和表2里有一项SNR和CR的指标,代码里没有给出是如何计算的.对于小目标增强,原文中提出For simplicity, we only use the best two super-resolved results of D3Dnet and MoCoPnet to perform detection.,这里是指后续的SNRG BSF SCRG CG四项指标只在best two super-resolved results上计算吗?根据提供的supp文件下的代码,是不是没有给出批量读取标注文件的接口

**Comment by XinyiYing** [comment]
公式6和公式9的分子就是SNR和CR
“For simplicity, we only use the best two super-resolved results of D3Dnet and MoCoPnet to perform detection.”这句话的意思是我只用D3Dnet和MoCoPnet在测试集上的超分辨结果进行小目标检测，随后计算检测的结果和原图之间的SNRG BSF SCRG CG四项指标，不是只选择了两张图片。
